lib/schemaUtils.js

```diff
@@ -134,19 +151,21 @@ function hash(input) {
 * @param {string} parameterSourceOption Specifies whether the schema being faked is from a request or response.
 * @param {*} components list of predefined components (with schemas)
 * @param {string} schemaFormat default or xml
-* @param {string} indentCharacter char for 1 unit of indentation
-* @param {object} schemaCache - object storing schemaFaker and schmeResolution caches
-* @param {object} stackLimit - The depth to which schema resolution will happen for nested refs
+* @param {object} schemaCache - object storing schemaFaker and schemaResolution caches
+* @param {object} options - a standard list of options that's globally passed around. Check options.js for more.
 * @returns {object} fakedObject
 */
 function safeSchemaFaker(oldSchema, resolveTo, resolveFor, parameterSourceOption, components,
-  schemaFormat, indentCharacter, schemaCache, stackLimit) {
+  schemaFormat, schemaCache, options) {
   var prop, key, resolvedSchema, fakedSchema,
     schemaResolutionCache = _.get(schemaCache, 'schemaResolutionCache', {}),
     schemaFakerCache = _.get(schemaCache, 'schemaFakerCache', {});
   let concreteUtils = components && components.hasOwnProperty('concreteUtils') ?
     components.concreteUtils :
     DEFAULT_SCHEMA_UTILS;
+  const indentCharacter = _.get(options, 'indentCharacter', '  '),
```

Comment:
Let's have single source of truth for default things. We can use `default` from options.js or make sure that these values are always overridden if empty in first place.

Reply:
As the options resolution comes before and this process sets the default if the options that are not present, i changed to only use the value directly as we do in other parts of the process.